src/napari/layers/utils/stack_utils.py

```diff
@@ -124,7 +124,9 @@ def split_channels(
                     allow_none=True,
                 )
             )
-        elif key == 'affine' and isinstance(val, np.ndarray):
+        elif key in ['rotate', 'shear'] or (
+            key == 'affine' and isinstance(val, np.ndarray)
+        ):
```

Comment:
Shouldn't the `isinstance(val, np.ndarray)` apply to all cases?

Reply:
My understanding from the comment is that key == affine that is not always ndarray. When it's Affine, it needs the special case. rotate and shear are always nd.array I think.

Reply:
right, got it 👍